{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HW 4.3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Run ChIPQC for:\n",
    "    https://drive.google.com/open?id=1dKLt_m_vnHjNVo-1qoACu9GZZC9OBa9v (Links to an external site.)Links to an external site.\n",
    "    EFF_H3K4Me1_1_srt_dupr_chr11.bam\n",
    "    EFF_H3K4Me1_2_srt_dupr_chr11.bam\n",
    "    EFF_Input_srt_dupr_chr11.bam (Control)\n",
    "- Sort and index bam file\n",
    "- Run MACS2\n",
    "- Create sample sheet\n",
    "- ChIPQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 19 Dec 2018 15:30:44: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:44: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:44: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:48: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:51: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:54: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:54: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:54: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:30:54: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:02: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:02: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:02: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:02: Done! \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:05: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:05: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:05: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:09: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:12: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:15: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:15: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:15: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:15: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:23: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:23: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:23: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:23: Done! \n",
      "[E::hts_hopen] Failed to open file EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam.bai\n",
      "[E::hts_open_format] Failed to open file EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam.bai\n",
      "samtools sort: can't open \"EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam.bai\": Exec format error\n",
      "INFO  @ Wed, 19 Dec 2018 15:31:24: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:24: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:24: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:28: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:30: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:33: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:33: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:33: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:33: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:41: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:42: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:46: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:49: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:52: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:52: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:52: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:31:52: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:00: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:00: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:00: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:00: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:32:01: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:01: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:01: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:04: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:07: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:09: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:09: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:09: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:09: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:18: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:18: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:18: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:18: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:32:19: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:19: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:19: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:22: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:25: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:28: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:28: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:28: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:28: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:36: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:36: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:36: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:36: Done! \n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:32:37: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:37: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:37: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:40: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:43: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:45: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:45: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:45: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:45: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:53: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:53: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:53: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:53: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:32:54: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:54: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:54: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:32:58: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #1 tag size is determined as 49 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #1 tag size = 49.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #1  total tags in treatment: 911074 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:00: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:03: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:03: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:03: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:03: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:11: #4 Write output xls file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:11: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:11: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_1_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:11: Done! \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:17: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:17: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:17: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:22:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:23: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:26: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:29: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:29: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:29: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:29: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:38: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:38: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:38: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:38: Done! \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:45: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:45: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:45: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:49:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:51: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:53: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:57: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:57: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:57: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:33:57: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:08: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:08: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:08: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:08: Done! \n",
      "[E::hts_hopen] Failed to open file EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam.bai\n",
      "[E::hts_open_format] Failed to open file EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam.bai\n",
      "samtools sort: can't open \"EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam.bai\": Exec format error\n",
      "INFO  @ Wed, 19 Dec 2018 15:34:10: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:10: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:10: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:15:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:17: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:20: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:23: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:23: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:23: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:23: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:33: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:34: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:34: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:34: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:34:35: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:35: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:35: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:39:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:41: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:43: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:46: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:46: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:46: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:46: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:56: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:56: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:56: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:56: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:34:57: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:57: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:34:57: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:02:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:03: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:05: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:09: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:09: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:09: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:09: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:19: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:19: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:19: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:19: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:35:20: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:20: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:20: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:25:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:26: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:29: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:33: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:33: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:33: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:33: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:46: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:46: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:46: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:46: Done! \n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:35:47: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:47: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:47: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:52:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:53: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:56: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:59: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:59: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:59: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:35:59: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:10: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:10: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:10: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:10: Done! \n",
      "[E::sam_parse1] missing SAM header\n",
      "[W::sam_read1] Parse error at line 1\n",
      "samtools sort: truncated file. Aborting\n",
      "INFO  @ Wed, 19 Dec 2018 15:36:11: \n",
      "# Command line: callpeak -t EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:11: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:11: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:16:  1000000 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:17: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #1  total tags in treatment: 1235840 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:20: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:23: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:23: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:23: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:23: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:33: #4 Write output xls file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:33: #4 Write broad peak in broadPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:33: #4 Write broad peak in bed12/gappedPeak format file... EFF_H3K4Me1_2_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:33: Done! \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:36: \n",
      "# Command line: callpeak -t EFF_Input_srt_dupr_chr11.sorted.bam -c EFF_Input_srt_dupr_chr11.bam -f BAM -n EFF_Input_srt_dupr_chr11.sorted.bam_MACS2 -g mm -p 1e-2 --broad --nomodel --bdg --shift 0 --extsize 147 --keep-dup all\n",
      "# ARGUMENTS LIST:\n",
      "# name = EFF_Input_srt_dupr_chr11.sorted.bam_MACS2\n",
      "# format = BAM\n",
      "# ChIP-seq file = ['EFF_Input_srt_dupr_chr11.sorted.bam']\n",
      "# control file = ['EFF_Input_srt_dupr_chr11.bam']\n",
      "# effective genome size = 1.87e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# pvalue cutoff for narrow/strong regions = 1.00e-02\n",
      "# pvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# qvalue will not be calculated and reported as -1 in the final output.\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is off\n",
      " \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:36: #1 read tag files... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:36: #1 read treatment tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:39: #1.2 read input tags... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #1 tag size is determined as 100 bps \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #1 tag size = 100.0 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #1  total tags in treatment: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #1  total tags in control: 594318 \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #1 finished! \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #2 Build Peak Model... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #2 Skipped... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #2 Use 147 as fragment length \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #3 Call peaks... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #3 Call broad peaks with given level1 -log10pvalue cutoff and level2: 2.00000, 1.00000... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:42: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:45: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:45: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... EFF_Input_srt_dupr_chr11.sorted.bam_MACS2_treat_pileup.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:45: #3   Write bedGraph files for control lambda (after scaling if necessary)... EFF_Input_srt_dupr_chr11.sorted.bam_MACS2_control_lambda.bdg \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:45: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:52: #4 Write output xls file... EFF_Input_srt_dupr_chr11.sorted.bam_MACS2_peaks.xls \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:52: #4 Write broad peak in broadPeak format file... EFF_Input_srt_dupr_chr11.sorted.bam_MACS2_peaks.broadPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:52: #4 Write broad peak in bed12/gappedPeak format file... EFF_Input_srt_dupr_chr11.sorted.bam_MACS2_peaks.gappedPeak \n",
      "INFO  @ Wed, 19 Dec 2018 15:36:52: Done! \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ChIP-seq.HW4.3/\n",
    "FILES=$(ls | grep \".bam\")\n",
    "for FILE in $FILES\n",
    "do\n",
    "    NAME=${FILE%%.*}.sorted.bam\n",
    "    samtools sort $FILE -o $NAME\n",
    "    samtools index $NAME\n",
    "    \n",
    "    macs2 callpeak \\\n",
    "    -t $NAME \\\n",
    "    -c EFF_Input_srt_dupr_chr11.bam \\\n",
    "    -f BAM \\\n",
    "    -n ${NAME}_MACS2 \\\n",
    "    -g mm \\\n",
    "    -p 1e-2 \\\n",
    "    --broad \\\n",
    "    --nomodel \\\n",
    "    --bdg \\\n",
    "    --shift 0 \\\n",
    "    --extsize 147 \\\n",
    "    --keep-dup all\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.Use Homer to search for TF motifs in broadPeak file (from Q1, EFF_H3K4Me1_1_srt_dupr_chr11.bam)\n",
    "genome reference: genome_chr11.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "source ~/.bash_profile\n",
    "cd ~/Documents/Dev/SRCourse/original/Applied-Bioinformatics/Unit2-module2-ChIPseq/ChIP-seq.HW4.3/\n",
    "KEYWORD=\"EFF_H3K4Me1_1_srt_dupr_chr11.sorted\"\n",
    "BED_NAME=$(ls | grep $KEYWORD | grep broadPeak)\n",
    "CHR11_GENOME=\"genome_chr11.fa\"\n",
    "OUT_NAME=${KEYWORD}_HomerMF\n",
    "ERR_NAME=${KEYWORD}_HomerMF.out\n",
    "\n",
    "findMotifsGenome.pl $BED_NAME $CHR11_GENOME $OUT_NAME &> $ERR_NAME"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### What are the top 5 knownResults?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ETV1\n",
      "Ets1-distal\n",
      "Fli1\n",
      "ERG\n",
      "ETS1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ChIP-seq.HW4.3/EFF_H3K4Me1_1_srt_dupr_chr11.sorted_HomerMF/\n",
    "RAW=$(head -6 knownResults.txt | cut -f 1 | sed '1d')\n",
    "for name in $RAW\n",
    "do\n",
    "    echo ${name%%(*}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### What is the difference between knownResults and homerResults?\n",
    "According to the [Manual](http://homer.ucsd.edu/homer/introduction/basics.html),  \n",
    "*De novo motif discovery allows you to directly query the sequence to discover which motifs are the MOST enriched sequences in your target set.  Known motif discovery will simply tell you which of the known motifs is most enriched in your target set.*\n",
    "\n",
    "The other thing is the reliability:  \n",
    "*The greatest advantage to using known motifs is found when you have a limited set of target sequences.  The less data that is available or the weaker the true signal, it is difficult for de novo motif finding to accurately define a signal that is significant.  Known motifs have the advantage of many less degrees of freedom and in may cases find the correct motifs when the enrichment falls below the 1e-10 thresholds for reliability when considering de novo results.*"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
